### PR TITLE
fix: redact Musixmatch usertoken from transport error strings

### LINKED ISSUE #431: security: Musixmatch usertoken leaks into last_error / failure-analysis screen
[mode: direct]
[model: sonnet]
[effort: medium]

## Description

The Musixmatch usertoken (a secret credential) can leak into `work_queue.last_error`, where it is then rendered verbatim on the failure-analysis screen and emitted to logs.

When a fetch fails at the transport layer (DNS failure, connection refused, TLS handshake error, context timeout), `internal/musixmatch.Client.FindLyrics` returns the error from `http.Client.Do` unmodified. Go wraps a transport failure in a `*url.Error` whose message embeds the full request URL. The usertoken is a query parameter on that URL, and `net/http`'s URL redaction only strips userinfo, not query parameters, so the raw token appears in the error string.

That error is stored verbatim as `work_queue.last_error` (`internal/queue`, `cause.Error()`), grouped into the failure "Reason" (`internal/reports`), and rendered on the failure-analysis UI. It is also logged via `slog(..., "error", err)` in the worker.

Deterministically reproduced: a transport error carrying a placeholder token surfaces the token in the returned error string.

## Steps to Reproduce

1. Configure `serve` mode with a Musixmatch usertoken.
2. Induce a transport-level failure to the upstream (e.g. block egress / no network) while the worker drains the queue.
3. Open the failure-analysis screen (or inspect `work_queue.last_error` / logs) for the failed rows.

## Expected Behavior

The credential is never written to `last_error`, rendered in the UI, or logged. The failure reason shows a redacted marker instead.

## Actual Behavior

The raw usertoken appears in the failure reason string (URL query `usertoken=...`).

## Environment

- Component: `internal/musixmatch`, surfaced via `internal/queue` + `internal/reports` + `web` failure-analysis UI
- Applies to `serve` mode on all platforms

## Additional Context

Fix (in progress): redact the token at the client boundary before returning transport / request-build errors, wrapping so `errors.Is`/`As` (context cancellation, circuit-breaker sentinels) are preserved.

Residual, follow-up considerations:
- Rows already persisted in `work_queue.last_error` before the fix may still contain the token; a scrub of existing rows and/or a render-time defense would cover historical data.
- Given the token may already have been exposed on the failure screen/logs, rotating the Musixmatch usertoken is prudent regardless of the code fix.

### PR DESCRIPTION
**Pull request by sydlexius**
## Summary

- The Musixmatch usertoken (a secret) could leak into `work_queue.last_error`, the failure-analysis screen, and logs. A transport-level failure from `http.Client.Do` surfaces as a `*url.Error` whose message embeds the full request URL; `net/http`'s URL redaction only strips userinfo, not query params, so the `usertoken=...` query parameter appeared verbatim and was returned/stored/rendered raw.
- Redact the token at the client boundary (`redactToken`) on the request-build and transport error paths -- the ones that echo the request URL. The wrapper overrides only the rendered message and delegates `Unwrap`, so `errors.Is`/`As` (context cancellation, circuit-breaker sentinels, `IsBenignMiss`) are unaffected. The percent-escaped token form is covered too.
- Reproduced deterministically and guarded by tests asserting the token is absent, the `REDACTED` marker is present, and the error chain is preserved.

## Linked issue

Closes #431

## Gates (run locally; CI enforces them)

- [x] `make gate` (race tests, patch coverage, golangci-lint, actionlint, govulncheck) -- all green
- [x] Patch coverage 100% on `redactToken` (added edge-case tests for nil / empty-token passthrough and escaped-token redaction)
- [x] Adversarial pre-push review -- confirmed no remaining leak path and error-chain preserved (verified against a real `*url.Error` from a live `httptest` timeout)

## Notes / follow-ups (tracked in #431)

- Rows already persisted in `work_queue.last_error` before this fix may still contain the token; a scrub of existing rows and/or a render-time defense would cover historical data.
- Because the token may already have been exposed, rotating the Musixmatch usertoken is prudent regardless of the code fix.

### CODE REVIEW

**Comment by sydlexius** [comment]
@coderabbitai approve

**Comment by sydlexius** [comment]
@coderabbitai approve

File: internal/musixmatch/client.go [hunk 1]
```diff
@@ -66,6 +66,45 @@ var (
...
+
+// redactedTokenError overrides only the rendered message, delegating identity
+// (Unwrap) to the wrapped error so errors.Is/As are unaffected.
+//
+// Note: because Unwrap exposes the original error, errors.As(err, &urlErr) still
+// yields a *url.Error whose .URL field holds the raw token. That is safe only so
+// long as no caller logs urlErr.URL directly; every current consumer renders via
+// Error() (the redacted string). A future caller that reads the struct field must
+// redact it itself.
+type redactedTokenError struct {
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply internal/musixmatch/client.go:91]: Fixed in 63d897f. Implemented fmt.GoStringer (`GoString() string { return e.Error() }`) so the %#v verb renders the redacted message instead of the raw struct fields; extended the test to assert %v/%+v/%s/%#v all keep the token redacted. errors.Is/As behavior is unchanged (Unwrap still delegates to the wrapped error).
